Burak improved div

myniqx opened the pull request:
![image](https://github.com/user-attachments/assets/a7963463-4f19-4cb3-8c38-8e8ed7082b02)
the idea is simple, desing your layout with that div block (which supports  most of css properties of div).  and add the custom components inside.

i try to implement a component which ll convert our old components to blocks dynamicly, but i cant figure those path issues. there is no way that i know to debug. (maybe you can check Component.tsx file out, see DynamicComponent and all those comment outted lines)

so my suggestion is if we go with payload admin panel to build pages, we should add block config to their path. so it can easy detect from payload. and some little adjustment to make them visible both frontend and payload admin panel.

this is the demo page's json:
```json
{
	"title": "Div Demo",
	"hero": {
		"type": "lowImpact",
		"links": []
	},
	"layout": [
		{
			"blockType": "div",
			"boxSettings": {
				"layout": "{\"alignItems\":\"stretch\",\"customGridTemplateColumns\":\"\",\"customGridTemplateRows\":\"\",\"flexDirection\":\"row\",\"flexWrap\":\"nowrap\",\"gap\":\"2rem\",\"gridTemplateColumns\":\"1fr 1fr\",\"gridTemplateRows\":\"1fr 1fr\",\"justifyContent\":\"flex-start\",\"type\":\"grid\"}",
				"size": "{\"width\":{\"preset\":\"100%\",\"unit\":\"vw\",\"usePreset\":true,\"value\":\"100\"},\"height\":{\"preset\":\"100%\",\"unit\":\"vh\",\"usePreset\":false,\"value\":\"60\"},\"minWidth\":{\"preset\":\"\",\"unit\":\"custom\",\"usePreset\":false,\"value\":\"\"},\"maxWidth\":{\"preset\":\"\",\"unit\":\"custom\",\"usePreset\":false,\"value\":\"\"},\"minHeight\":{\"preset\":\"\",\"unit\":\"custom\",\"usePreset\":false,\"value\":\"\"},\"maxHeight\":{\"preset\":\"\",\"unit\":\"custom\",\"usePreset\":false,\"value\":\"\"}}",
				"margin": "{\"colors\":{},\"widths\":{\"bottom\":\"12\",\"left\":\"12\",\"right\":\"12\",\"top\":\"12\"}}",
				"padding": "{\"colors\":{},\"widths\":{\"bottom\":\"12\",\"left\":\"12\",\"right\":\"12\",\"top\":\"12\"}}",
				"border": "{\"colors\":{},\"widths\":{\"bottom\":\"3\",\"left\":\"3\",\"right\":\"3\",\"top\":\"3\"}}",
				"borderRadius": "1rem;1rem;1rem;1rem",
				"backgroundColor": "{\"angle\":90,\"colors\":[{\"color\":\"#932121\",\"position\":0},{\"color\":\"#987171\",\"position\":100}],\"position\":\"center\",\"type\":\"radial-gradient\"}",
				"children": [
					{
						"blockType": "div_level_1",
						"boxSettings": {
							"layout": "{\"alignItems\":\"stretch\",\"customGridTemplateColumns\":\"\",\"customGridTemplateRows\":\"\",\"flexDirection\":\"row\",\"flexWrap\":\"nowrap\",\"gap\":\"0\",\"gridTemplateColumns\":\"none\",\"gridTemplateRows\":\"none\",\"justifyContent\":\"flex-start\",\"type\":\"none\"}",
							"backgroundColor": "{\"angle\":90,\"colors\":[{\"color\":\"#ffffff\",\"position\":0},{\"color\":\"#000000\",\"position\":100}],\"position\":\"center\",\"type\":\"radial-gradient\"}",
							"children": []
						},
						"id": "67dd73a9000cef9a7cd1c288"
					},
					{
						"blockType": "div_level_1",
						"boxSettings": {
							"layout": "{\"alignItems\":\"stretch\",\"customGridTemplateColumns\":\"\",\"customGridTemplateRows\":\"\",\"flexDirection\":\"row\",\"flexWrap\":\"nowrap\",\"gap\":\"0\",\"gridTemplateColumns\":\"none\",\"gridTemplateRows\":\"none\",\"justifyContent\":\"flex-start\",\"type\":\"none\"}",
							"backgroundColor": "{\"angle\":90,\"colors\":[{\"color\":\"#ffffff\",\"position\":0},{\"color\":\"#000000\",\"position\":100}],\"position\":\"center\",\"type\":\"conic-gradient\"}",
							"children": []
						},
						"id": "67dd73bb000cef9a7cd1c28a"
					},
					{
						"blockType": "div_level_1",
						"boxSettings": {
							"layout": "{\"alignItems\":\"stretch\",\"customGridTemplateColumns\":\"\",\"customGridTemplateRows\":\"\",\"flexDirection\":\"row\",\"flexWrap\":\"nowrap\",\"gap\":\"0\",\"gridTemplateColumns\":\"none\",\"gridTemplateRows\":\"none\",\"justifyContent\":\"flex-start\",\"type\":\"none\"}",
							"backgroundColor": "{\"angle\":90,\"colors\":[{\"color\":\"#ffffff\",\"position\":0},{\"color\":\"#000000\",\"position\":100}],\"position\":\"center\",\"type\":\"linear-gradient\"}",
							"children": []
						},
						"id": "67dd73c9000cef9a7cd1c28c"
					},
					{
						"blockType": "div_level_1",
						"boxSettings": {
							"layout": "{\"alignItems\":\"stretch\",\"customGridTemplateColumns\":\"\",\"customGridTemplateRows\":\"\",\"flexDirection\":\"row\",\"flexWrap\":\"nowrap\",\"gap\":\"0\",\"gridTemplateColumns\":\"none\",\"gridTemplateRows\":\"none\",\"justifyContent\":\"flex-start\",\"type\":\"none\"}",
							"backgroundColor": "{\"angle\":90,\"colors\":[{\"color\":\"#ffffff\",\"position\":0},{\"color\":\"#c21a1a\",\"position\":50},{\"color\":\"#000000\",\"position\":100}],\"position\":\"center\",\"type\":\"radial-gradient\"}",
							"children": []
						},
						"id": "67dd73d7000cef9a7cd1c28e"
					}
				]
			},
			"id": "67dd7324000cef9a7cd1c286"
		}
	],
	"meta": {},
	"publishedAt": "2025-03-21T14:09:03.400Z",
	"slug": "div-demo",
	"slugLock": true,
	"updatedAt": "2025-03-21T14:20:50.837Z",
	"createdAt": "2025-03-21T13:59:35.894Z",
	"_status": "draft",
	"id": "67dd70c73cff25c0d5072b91"
}
```

7alip commented on apps/hvn/src/blocks/Component/componentList.ts:
```diff
@@ -0,0 +1,9 @@
+export const componentOptions = [
+  { label: 'application-from (../../components/application-from)', value: '../../components/application-from' },
```
How do you think to pass data to these components? We should be able to pass Payload data to the component props.

myniqx replied:
i havent thought until u ask. we may figure it out on the way. which prop we might need. but i dont think we need alot. since we want to make site static as much as it can be, we can use SSR components and fetch the data what we need for every component inside of it..

7alip commented on apps/hvn/src/blocks/Component/Component.tsx:
```diff
@@ -0,0 +1,25 @@
+  ); */
+
+  return (
+    <React.Suspense fallback={<div>Yükleniyor...</div>}>
```
Static components should be rendered on the server side (RSC), as most pages will be static and can be revalidated upon updates. This approach also benefits SEO.

You might be concerned about performance here. You can investigate further, but if the pages are cached, importing all possible components without dynamic imports may not be an issue.

myniqx replied:
fair point.

myniqx commented:
@7alip 
i removed divblock from config (not its codes, those codes can help us).
i added container block (section/container) for start.
i added a guide how to register components to payload (a .md file)
we can now open some issues to create new component with block configs and some issues to create config file for current components.

7alip commented:
@myniqx 

<img width="1311" alt="Screenshot 2025-04-08 at 17 23 52" src="https://github.com/user-attachments/assets/0afbe714-ecc4-472b-aca0-ee5413ffb2aa" />

myniqx commented:
- moved all payload-related things to payload-config 
- moved all components to fc/ui
- moved all common utils to fc/ui/libs
- fixed all type-check error
